### PR TITLE
Keep top level nb_visits_converted metric

### LINKED ISSUE #10109: On email reports the conversion rate is zero always
Hi
after receiving daily reports , on "all referrer " part , the last column is always zero 

3845fb3bb1b1dc333ff83500ba6579557fb31830_1_690x183

### PR DESCRIPTION
**Pull request by diosmosis**
Currently we delete this metric "because it could differ from the sum of each goal conversions", but it should, since nb_visits_converted is not the sum of each goal's conversions. Conversions can happen multiple times per visit, so this would not be accurate.

Also, it is not the sum of each goals' nb_visits_converted, since visits can share conversions for different goals. Keeping it puts the correct value of conversion_rate in scheduled reports.

Fixes #10109

### CODE REVIEW

File: plugins/CoreHome/Columns/Metrics/ConversionRate.php [hunk 1]
```diff
@@ -43,8 +44,20 @@ public function format($value, Formatter $formatter)
...
     {
-        $nbVisitsConverted = $this->getMetric($row, 'nb_visits_converted');
         $nbVisits = $this->getMetric($row, 'nb_visits');
+        $nbVisitsConverted = $this->getMetric($row, 'nb_visits_converted');
+        if ($nbVisitsConverted === false) {
+            $goals = $this->getMetric($row, 'goals');
+
+            $nbVisitsConverted = 0;
+            foreach ($goals as $idGoal => $values) {
+                if ($idGoal === 0 || $idGoal === '0' || $idGoal == 'idgoal=0') {
```
Comment (tsteur) [inline plugins/CoreHome/Columns/Metrics/ConversionRate.php]: would this also need to be done for `goals = -1`?
@mattab maybe you can have a look at the PR some time I actually don't quite understand if the behaviour is expected.

Comment (diosmosis) [reply plugins/CoreHome/Columns/Metrics/ConversionRate.php]: yeah, that makes sense don't want to count "conversions" for abandoned carts here

File: plugins/CoreHome/Columns/Metrics/ConversionRate.php [hunk 2]
```diff
@@ -43,8 +44,28 @@ public function format($value, Formatter $formatter)
...
         $nbVisits = $this->getMetric($row, 'nb_visits');
+        $nbVisitsConverted = $this->getMetric($row, 'nb_visits_converted');
+        if ($nbVisitsConverted === false) {
+            $goals = $this->getMetric($row, 'goals');
+
+            if (!empty($goals)) {
+                $nbVisitsConverted = 0;
+                foreach ($goals as $idGoal => $values) {
+                    if ($idGoal === 0
+                        || $idGoal === '0'
```
Comment (mattab) [inline plugins/CoreHome/Columns/Metrics/ConversionRate.php]: is there maybe a CONST we could reference instead of hard coding the value?

Comment (diosmosis) [reply plugins/CoreHome/Columns/Metrics/ConversionRate.php]: Can do

**Comment by mattab** [comment]
Feedback
* in general i'm not sure if this is accurate. "Nb visits converted" should be number of visits with at least one converted goal. but this logic takes the "max" value of "nb visits converted" for any given goal. The "Real" accurate value of "Nb visits converted" will be equal or higher to the value processed here. For example we may have 5 visits converted to goal X and 10 visits converted to goal Y, this PR will say that 10 visits are converted, but it could be that anywhere between 10 and 15 visits are converted. So it wouldn't show accurate numbers. Is there maybe another way to have this accurate nb_visits_converted metric?
* A sub-module change is included in the PR which I suppose is not expected

**Comment by diosmosis** [comment]
Moving to 3.11 since it's not a straightforward fix

**Comment by diosmosis** [comment]
@mattab removed the code you were talking about (that was for showing the conversion rate for old reports, but of course that won't work).

From testing manually and looking at automated tests, the nb_visits_converted metric that's not in the `goals` array looks accurate. It's computed by `queryVisitsByDimension()` which aggregates over the log_visit table, so I'm not sure why it would be inaccurate. Don't think there's an issue in keeping it.

**Comment by mattab** [comment]
LGTM 
(tests need updated before merge)

**Comment by tsteur** [comment]
Some tests are still failing @diosmosis . Feel free to merge once the build is green

**Pull request merged**